R/ebayesthresh.R

```diff
@@ -17,31 +18,39 @@ function(x, prior = "laplace", a = 0.5, bayesfac = FALSE, sdev = NA, verbose = F
 #   it is estimated using the function mad(x).
 #
 #  find the standard deviation if necessary and estimate the parameters
-	if(is.na(sdev)) sdev <- mad(x, center = 0)
-	x <- x/sdev
+  if(length(sdev)==1){
+  	if(is.na(sdev)) sdev <- mad(x, center = 0)
+  } else{
+    if(length(sdev)!=length(x)) stop("Standard deviation has to be homogeneous or has the same length as x.")
+  }
+
+  m_sdev <- mean(sdev)
+  s <- sdev/m_sdev
+  x <- x/m_sdev
```

Comment:
I'm guessing that it is so that the code also works as before for cauchy prior.... Can you check the result is the same for Laplace prior whether or not you do this normalization?